test/erc721m.test.ts

```diff
@@ -503,6 +503,33 @@ describe('ERC721M', function () {
       await expect(mint).to.be.revertedWith('NotEnoughValue');
     });
 
+    it('revert on reentrancy', async () => {
+      const reentrancyFactory = await ethers.getContractFactory(
+        'TestReentrantExploit',
+      );
+      const reentrancyExploiter = await reentrancyFactory.deploy(
+        contract.address,
+      );
+      await reentrancyExploiter.deployed();
+      await contract.setStages([
+        {
+          price: ethers.utils.parseEther('0.1'),
+          walletLimit: 0,
+          merkleRoot: ethers.utils.hexZeroPad('0x', 32),
+          maxStageSupply: 0,
+          startTimeUnixSeconds: 0,
+          endTimeUnixSeconds: 100000,
+        },
+      ]);
+      await contract.setMintable(true);
+
+      await expect(
+        reentrancyExploiter.exploit(1, [], 0, '0x', {
```

Comment:
Can you help me understand the sequence?
erc721M -> exploiter.explit -> erc721M.mint -> exploiter.onERC721Received -> erc721M.mint <-- fail for reentrancy

Reply:
yes this is exactly the sequence